### PR TITLE
Locations cleanup

### PR DESCRIPTION
**Pull request by skriss**
This PR does some minor cleanup on the backup-locations work.  The main change is to standardize/simplify how controllers construct a plugin manager.  They all now take a `newPluginManager` function pointer as an argument, that only requires a logger to be passed at execution time.

### CODE REVIEW

**Comment by nrb** [comment]
Is this to aid in testing the controllers? That seems to be the biggest change that I see looking over it.

**Comment by skriss** [comment]
Well, passing in a `newPluginManager` function pointer is to make testing easier, so that function can be replaced with a test fake - but we were already doing that.  In the current state, though, we're passing a `pluginRegistry` and `logLevel` to most/all of the controllers too, but they're usually only being used as arguments to pass to the `newPluginManager` func.  This change constructs a closure, in `server.go`, that encloses a `pluginRegistry` and `logLevel` -- so now the controllers don't need to take those things as arguments, and the `newPluginManager` signature is simpler.  

tldr -- simplification and standardization.

**Comment by skriss** [comment]
To clarify, the controllers already had a `newPluginManager` function pointer field, which we were setting in the constructors, reason being for testing.  This PR passes it in, as a closure.
